homeassistant/components/sensor/co2signal.py

```diff
@@ -0,0 +1,131 @@
+"""
+Support for the CO2signal platform.
+For more details about this platform, please refer to the documentation
+"""
+import logging
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+
+from homeassistant.const import (
+    CONF_TOKEN, CONF_LATITUDE, CONF_LONGITUDE)
+from homeassistant.helpers.config_validation import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import Entity
+
+CONF_COUNTRY_CODE = "country_code"
+
+REQUIREMENTS = ['co2signal==0.4.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTRIBUTION = 'Data provided by CO2signal'
+
+msg_location = "Please use either coordinates or the country code. " \
+               "For the coordinates, " \
+               "you need to use both latitude and longitude."
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_TOKEN): cv.string,
+    vol.Inclusive(CONF_LATITUDE, 'coords', msg=msg_location): cv.latitude,
+    vol.Inclusive(CONF_LONGITUDE, 'coords', msg=msg_location): cv.longitude,
+    vol.Optional(CONF_COUNTRY_CODE, default='None'): cv.string,
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the CO2signal sensor."""
+
+    _LOGGER.debug("Setting up the CO2signal platform")
+
+    token = config.get(CONF_TOKEN)
+    lat = config.get(CONF_LATITUDE, hass.config.latitude)
+    lon = config.get(CONF_LONGITUDE, hass.config.longitude)
+    country_code = config.get(CONF_COUNTRY_CODE)
+
+    if country_code != 'None':
+        location_type = 'country_code'
+    else:
+        location_type = 'coordinates'
+
+    _LOGGER.debug("Setting up the sensor using the %s", location_type)
+
+    devs = []
+
+    devs.append(CO2Sensor(token,
+                          country_code,
+                          lat,
+                          lon,
+                          location_type))
+
+    add_entities(devs, True)
+
+
+class CO2Sensor(Entity):
+    """Implementation of the CO2Signal sensor."""
+
+    def __init__(self, token, country_code, lat, lon,
+                 location_type):
+        """Initialize the sensor."""
+
+        self._token = token
+        self._country_code = country_code
+        self._latitude = lat
+        self._longitude = lon
+        self._location_type = location_type
+        self._unit = 'CO2eq/kWh'
```

Comment:
This could be extracted to a constant at the module level. Then we don't need the instance attribute but can use the constant directly.